account_statement_import_fr_cfonb/wizard/account_statement_import.py

```diff
@@ -0,0 +1,206 @@
+# Copyright 2014-2020 Akretion (http://www.akretion.com)
+# @author Alexis de Lattre <alexis.delattre@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+from datetime import datetime
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+CFONB_WIDTH = 120
+
+
+class AccountStatementImport(models.TransientModel):
+    _inherit = "account.statement.import"
+
+    _excluded_accounts = []
+
+    def _parse_cfonb_amount(self, amount_str, nb_of_dec):
+        """Taken from the cfonb lib"""
+        amount_str = amount_str[:-nb_of_dec] + "." + amount_str[-nb_of_dec:]
+        # translate the last char and set the sign
+        credit_trans = {
+            "A": "1",
+            "B": "2",
+            "C": "3",
+            "D": "4",
+            "E": "5",
+            "F": "6",
+            "G": "7",
+            "H": "8",
+            "I": "9",
+            "{": "0",
+        }
+        debit_trans = {
+            "J": "1",
+            "K": "2",
+            "L": "3",
+            "M": "4",
+            "N": "5",
+            "O": "6",
+            "P": "7",
+            "Q": "8",
+            "R": "9",
+            "}": "0",
+        }
+        assert (
+            amount_str[-1] in debit_trans or amount_str[-1] in credit_trans
+        ), "Invalid amount in CFONB file"
+        if amount_str[-1] in debit_trans:
+            amount_num = float("-" + amount_str[:-1] + debit_trans[amount_str[-1]])
+        elif amount_str[-1] in credit_trans:
+            amount_num = float(amount_str[:-1] + credit_trans[amount_str[-1]])
+        return amount_num
+
+    def _check_journal_bank_account(self, journal, account_number):
+        res = super()._check_journal_bank_account(journal, account_number)
+        if not res:
+            # compare the bank account number only instead of the full IBAN
```

Comment:
ne comparer uniquement si c'est un compte français.
```python
if jrnl_full_acc_number[0:2] != 'FR':
    return res
```

Reply:
I removed that code. It's now handled in account_statement_import.